### PR TITLE
CRM-18106 create activity against contact deleted by merge

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Note that in discussion on the ticket & IRC we talked about storing more data in the activity details - I'm treating that as a potential stage 2 at the moment (and investigating logging reversals)

---
- [CRM-18106: Add activity to deleted contacts showing they were deleted by merge](https://issues.civicrm.org/jira/browse/CRM-18106)

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Dedupe/Merger.php [hunk 1]
```diff
@@ -1930,4 +1921,28 @@ public static function addCustomTablesExtendingContactsToCidRefs(&$cidRefs) {
...
+   */
+  public static function createMergeActivities($mainId, $otherId) {
+    $activity = civicrm_api3('activity', 'create', array(
+      'subject' => ts('Contact ID %1 has been merged and deleted.', array(1 => $otherId)),
+      'target_contact_id' => $mainId,
+      'activity_type_id' => 'Contact Merged',
+      'status_id' => 'Completed',
+    ));
+    civicrm_api3('activity', 'create', array(
+      'subject' => ts('Contact ID has been merged and deleted into %1.', array(1 => $mainId)),
```
Comment (seamuslee001) [inline CRM/Dedupe/Merger.php]: 'Should this not read Contact has been deleted and merged into %1' ?

Comment (eileenmcnaughton) [inline CRM/Dedupe/Merger.php]: Surely you are not telling me I'm losing the plot?

Comment (seamuslee001) [inline CRM/Dedupe/Merger.php]: maybe :-P and  I am serious but don't call me Shirley

**Comment by eileenmcnaughton** [comment]
Howzat?

**Comment by seamuslee001** [comment]
excellent :-)

**Pull request merged**